### PR TITLE
3.28.0 Changelog

### PR DESCRIPTION
**Pull request by c2thorn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,24 @@
 ## 3.28.0 (Unreleased)
+
+FEATURES:
+* **New Data Source:** `google_redis_instance` ([#2209](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2209))
+* **New Resource:** `google_notebook_environment` ([#2199](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2199))
+* **New Resource:** `google_notebook_instance` ([#2199](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2199))
+
+IMPROVEMENTS:
+* Add SHARED_LOADBALANCER_VIP as option for google_compute_address.purpose ([#2204](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2204))
```
Comment (rileykarson) [inline CHANGELOG.md:9]: I think this note needs touching up

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+* app_engine: added the option to specify `database_type` in `google_app_engine_application` ([#2193](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2193))
+* compute: Added `mode` to `google_compute_autoscaler` `autoscaling_policy` ([#2214](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2214))
+* compute: Added `remove_instance_state_on_destroy` to `google_compute_per_instance_config` to control deletion of underlying instance state. ([#2187](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2187))
+* compute: Added `remove_instance_state_on_destroy` to `google_compute_region_per_instance_config` to control deletion of underlying instance state. ([#2187](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2187))
+* compute: Added `scale_down_control` for `google_compute_autoscaler` `autoscaling_policy` ([#2214](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2214))
+* dns: enabled `google_dns_policy` to accept network id #6577 ([#2189](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2189))
```
Comment (danawillow) [inline CHANGELOG.md:15]:
```suggestion
* dns: enabled `google_dns_policy` to accept network id ([#2189](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2189))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,24 @@
...
+
+BUG FIXES:
+* appengine: Added polling to `google_app_engine_firewall_rule` to prevent issues with eventually consistent creation ([#2197](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2197))
```
Comment (danawillow) [inline CHANGELOG.md:18]: This says "appengine" but an enhancement above says "app_engine". let's change both to be whatever we've done most often in the past

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,24 @@
 ## 3.28.0 (Unreleased)
+
+FEATURES:
+* **New Data Source:** `google_redis_instance` ([#2209](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2209))
+* **New Resource:** `google_notebook_environment` ([#2199](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2199))
+* **New Resource:** `google_notebook_instance` ([#2199](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2199))
+
+IMPROVEMENTS:
+* compute: Added SHARED_LOADBALANCER_VIP as an option for `google_compute_address.purpose` ([#2204](https://github.com/terraform-providers/terraform-provider-google-beta/pull/2204))
```
Comment (danawillow) [inline CHANGELOG.md:9]: Can this entry be moved to be with its compute neighbors?

Comment (c2thorn) [reply CHANGELOG.md]: I forgot my abc's

**Pull request merged**